src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -563,27 +624,231 @@ internal SafeFileHandle OpenPort(string portName)
                 _handle = null;
                 throw;
             }
-        }
 
-        ~SerialStream()
-        {
-            Dispose(false);
+            _ioLoop = Task.Factory.StartNew(IOLoop, TaskCreationOptions.LongRunning);
```

Comment:
This should use an overload that let's you specify TaskScheduler.Default as the TaskScheduler; otherwise, in the (likely rare) case where someone accessed this on another scheduler, this task will be queued to that scheduler instead of creating a new thread.